### PR TITLE
feature: vectorize knowledge horizon function calls, and add some todos

### PR DESCRIPTION
**Pull request by Flix6x**
Much needed vectorization.

- [x] Need to resolve some todos.
- [x] Want to add tests for ex_post and ex_ante knowledge horizon functions. Especially the ex_post, because it is the most common. We're only testing the most difficult ones right now.

### CODE REVIEW

File: timely_beliefs/tests/test_belief_query.py [hunk 1]
```diff
@@ -56,13 +56,13 @@ def test_query_belief_for_sensor_with_unique_knowledge_time(
 
 
 @pytest.fixture(scope="function")
```
Comment (victorgarcia98) [inline timely_beliefs/tests/test_belief_query.py:58]: Probably, this fixture should go in conftest.py.

File: timely_beliefs/tests/test_belief_query.py [hunk 2]
```diff
@@ -187,7 +187,7 @@ def test_query_belief_by_belief_time(
         len(
```
Comment (victorgarcia98) [inline timely_beliefs/tests/test_belief_query.py:187]: Isn't `len(bdf.index) == 0` equivalent to `bdf.empty`?

Comment (victorgarcia98) [reply timely_beliefs/tests/test_belief_query.py:187]: Also, I think the test could be simplified using a for loop to test for different `belief_time` or the use of `pytest.parametrize`.

File: timely_beliefs/tests/test_belief_query.py [hunk 3]
```diff
@@ -430,17 +430,17 @@ def test_select_most_recent_deterministic_beliefs(
 
 
 def test_select_most_recent_probabilistic_beliefs(
-    ex_post_time_slot_sensor: DBSensor,
-    multiple_probabilistic_day_ahead_beliefs_about_ex_post_time_slot_event: list[
+    ex_ante_economics_sensor: DBSensor,
```
Comment (victorgarcia98) [inline timely_beliefs/tests/test_belief_query.py:433]: Not sure about this, but I think that  `economics_sensor` should be `economic_sensor`, denoting that it's a sensor of an economic nature.

**Review by victorgarcia98 — COMMENTED** [review]
Cool!

Just some ideas to simplify the tests.

File: timely_beliefs/tests/test_belief_query.py (hunk 3 above)
Comment (Flix6x) [reply timely_beliefs/tests/test_belief_query.py:385]: I kinda liked how the words *economics* and *physics* fit together. It's only a test variable name anyway.

File: timely_beliefs/tests/test_belief_query.py (hunk 1 above)
Comment (Flix6x) [reply timely_beliefs/tests/test_belief_query.py:58]: The beliefs data set up here is only used in the scope of this module, so I think it's the right place. Fixtures don't *have* to be in conftests per se.

**Comment by Flix6x** [comment]
Thanks for the review. Good idea to refactor that test using `pytest.mark.parametrize`. It looks much cleaner now.

**Review by victorgarcia98 — APPROVED** [review]

**Pull request merged**